main.cpp

```diff
@@ -56,46 +56,83 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* name);   //1
+    int value; //2
+    std::string name; //3
 };
 
-struct <#structName1#>                                //4
+T::T(int v, const char* myName) :
+value(v),
+name(myName)
+{}
+
+struct CompareBear                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    auto* compare(T* a, T* b) //5
```

Comment:
Why would you make this return `auto*`?  
```
 5) provide proper types for the arguments to compare, and a proper return value
         be sure to make it work with the pointer semantics being used in the if statements.
         see 8) for a hint about what type to use.
```
`a` and `b` are the same type.
The first two lines in this function are returning either `a` or `b`.
Surely that would hint that the return type should be the same as the type of `a` or `b`, no? 
- [ ] revise